addons/base_vat/models/res_partner.py

```diff
@@ -150,7 +150,7 @@ def fix_eu_vat_number(self, country_id, vat):
                 vat = country_code + vat
         return vat
 
-    @api.constrains('vat')
+    @api.constrains('vat', 'commercial_partner_id.country_id.code')
```

Comment:
isn't just `commercial_partner_id.country_id` enough?

Reply:
In fact country_id should be enough

Reply:
> isn't just `commercial_partner_id.country_id` enough?

Just for Info: 

`@constrains only supports simple field names, dotted names (fields of relational fields e.g. partner_id.customer) are not supported and will be ignored`
See: https://github.com/odoo/odoo/blob/12.0/odoo/api.py#L156